commit 4e27c9d56ccbc6ed6f6f576c8a2d498647938c4a
Author: Stephan Hartmann <stha09@googlemail.com>
Date:   Fri Jan 13 17:57:44 2023 +0000

    libstdc++: fix narrowing in blink::DarkModeLABColorSpace
    
    Clang errors out with narrowing from double to float. Simplify to std::cbrt and x*x*x instead.
    
    Bug: 957519
    Change-Id: Ife17cca478a95eccd582e87f4e9ae5420e72321b
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4164980
    Reviewed-by: Jeremy Roman <jbroman@chromium.org>
    Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
    Cr-Commit-Position: refs/heads/main@{#1092495}

diff --git a/third_party/blink/renderer/platform/graphics/dark_mode_lab_color_space.h b/third_party/blink/renderer/platform/graphics/dark_mode_lab_color_space.h
index 999c3e5b58ae0..28a15cd7f9485 100644
--- a/third_party/blink/renderer/platform/graphics/dark_mode_lab_color_space.h
+++ b/third_party/blink/renderer/platform/graphics/dark_mode_lab_color_space.h
@@ -125,8 +125,7 @@ class DarkModeLABColorSpace {
   // https://en.wikipedia.org/wiki/CIELAB_color_space#Reverse_transformation.
   SkV3 FromXYZ(const SkV3& v) const {
     auto f = [](float x) {
-      return x > kSigma3 ? pow(x, 1.0f / 3.0f)
-                         : x / (3 * kSigma2) + 4.0f / 29.0f;
+      return x > kSigma3 ? std::cbrt(x) : x / (3 * kSigma2) + 4.0f / 29.0f;
     };
 
     float fx = f(v.x / kIlluminantD50.x);
@@ -145,7 +144,7 @@ class DarkModeLABColorSpace {
   // https://en.wikipedia.org/wiki/CIELAB_color_space#Forward_transformation.
   SkV3 ToXYZ(const SkV3& lab) const {
     auto invf = [](float x) {
-      return x > kSigma ? pow(x, 3.0f) : 3.0f * kSigma2 * (x - 4.0f / 29.0f);
+      return x > kSigma ? x * x * x : 3.0f * kSigma2 * (x - 4.0f / 29.0f);
     };
 
     SkV3 v = {Clamp(lab.x, 0.0f, 100.0f), Clamp(lab.y, -128.0f, 128.0f),
